src/main/java/org/team1540/robot2024/subsystems/vision/AprilTagVisionIOMegaTag2.java

```diff
@@ -0,0 +1,63 @@
+package org.team1540.robot2024.subsystems.vision;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.DriverStation;
+import org.team1540.robot2024.util.vision.LimelightHelpers;
+
+import java.util.Arrays;
+import java.util.function.Supplier;
+
+public class AprilTagVisionIOMegaTag2 implements AprilTagVisionIO {
+    private final String name;
+    private final Supplier<Rotation2d> heading;
+
+    public AprilTagVisionIOMegaTag2(String name, Pose3d cameraOffsetMeters, Supplier<Rotation2d> heading) {
+        this.name = name;
+        this.heading = heading;
+        LimelightHelpers.setCameraMode_Processor(name);
+        LimelightHelpers.setLEDMode_PipelineControl(name);
+        setPoseOffset(cameraOffsetMeters);
+    }
+
+    @Override
+    public void updateInputs(AprilTagVisionIOInputs inputs) {
+        LimelightHelpers.SetRobotOrientation(name, heading.get().getDegrees(), 0, 0, 0, 0, 0);
+        LimelightHelpers.PoseEstimate measurement =
+                DriverStation.isDisabled()
+                        ? LimelightHelpers.getBotPoseEstimate_wpiBlue(name)
+                        : LimelightHelpers.getBotPoseEstimate_wpiBlue_MegaTag2(name);
+        Pose3d limelightPose = new Pose3d(measurement.pose);
+        inputs.estimatedPoseMeters = new Pose3d(
+                limelightPose.getX(),
+                limelightPose.getY() + 0.105,
+                limelightPose.getZ(),
+                limelightPose.getRotation());
+        inputs.lastMeasurementTimestampSecs = measurement.timestampSeconds;
+        inputs.numTagsSeen = measurement.tagCount;
+        inputs.seenTagIDs = Arrays.stream(measurement.rawFiducials).mapToInt(fiducial -> fiducial.id).toArray();
+        inputs.avgTagDistance = measurement.avgTagDist;
+    }
+
+    @Override
+    public void setPoseOffset(Pose3d poseOffsetMeters) {
+        LimelightHelpers.setCameraPose_RobotSpace(
+                name,
+                poseOffsetMeters.getX(),
+                poseOffsetMeters.getY(),
+                poseOffsetMeters.getZ(),
+                Math.toDegrees(poseOffsetMeters.getRotation().getX()),
+                Math.toDegrees(poseOffsetMeters.getRotation().getY()),
+                Math.toDegrees(poseOffsetMeters.getRotation().getZ()));
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public void takeSnapshot(String snapshotName) {
```

Comment:
Where do snapshots actually go?

Reply:
Limelight storage, but I believe it fills up pretty quick and we kept forgetting to clear the storage so this wasn't very useful.

Reply:
Do you access them through the web gui somewhere, and would adding a usb drive be useful for this in the future?

Reply:
They are availible in the web gui, and I don't think that a usb drive works on a limelight, I don't think the place they go is configureable. We shouldn't be taking snapshots every second anyways, it would probably be more helpful if we just took a snapshot every time we made a shot or something.